### PR TITLE
Export equivalent circuit Q3D/Q2D

### PR DESCRIPTION
**Pull request by gmalinve**
Export equivalent circuit Q3D/Q2D

@maxcapodi78 @MaxJPRey @Samuelopez-ansys 
there's a part of the method ExportCircuit not implemented.
It's the include Chip Package Control (include_cpp bool), I couldn't get my head around how to extract this info and apparently is not even accessible so I left it False. Maybe we should state it somewhere.
I'm going to open a defect in tfs for this.
image
image

### CODE REVIEW

File: pyaedt/q3d.py [hunk 1]
```diff
@@ -709,6 +709,425 @@ def export_matrix_data(
...
+        if rise_time_unit:
+            if rise_time_unit not in ["fs", "ps", "ns", "us", "ms", "s", "min", "hour", "day"]:
+                self.logger.error("Invalid rise time unit.")
+                return False
+        else:
+            rise_time_unit = "s"
+
+        rise_time = rise_time_value + rise_time_unit
+
+        if file_type.lower() not in ["hspice", "welement", "rlcg"]:
```
Comment (MaxJPRey) [inline pyaedt/q3d.py:1047]:
```suggestion
        if file_type.lower() not in ["hspice", "welement", "rlgc"]:
```

File: pyaedt/q3d.py [hunk 2]
```diff
@@ -709,6 +709,425 @@ def export_matrix_data(
...
+                    self.logger.error("Invalid resistance unit.")
+                    return False
+
+            coupling_limits.append("ResLimit:=")
+            coupling_limits.append(res_limit)
+        else:
+            coupling_limit_value = "None"
+            coupling_limits.append(coupling_limit_value)
+
+        # if include_cpp and not [x for x in [include_dcr, include_acr, include_dcl, include_acl, add_resistance] if x]:
```
Comment (MaxJPRey) [inline pyaedt/q3d.py:1008]: Should we uncomment this section, otherwise `include_cpp` is never used?

File: pyaedt/q3d.py [hunk 3]
```diff
@@ -709,6 +709,425 @@ def export_matrix_data(
...
+        freq=0,
+        file_type="HSPICE",
+    ):
+        """Export matrix data.
+
+        Parameters
+        ----------
+        file_name : str
+            Full path to save the equivalent circuit to.
+            Options for file extensions are: .cir, .sml, .sp, .pkg, .spc, .lib, .ckt, .bsp, .dml, .icm.
```
Comment (PipKat) [inline pyaedt/q3d.py:752]:
```suggestion
            Options for file extensions are CIR, SML, SP, PKG, SPC, LIB, CKT, BSP, DML, and ICM.
```

Comment (PipKat) [reply pyaedt/q3d.py:752]: Google Developer Style Guide, which we are following, and also the Ansys Style Guide, says to use all capital letters in reference to file extensions when not using an entire file name (such as ``MatrixData1.cir``, which is formatted as a literal string.

File: pyaedt/q3d.py [hunk 4]
```diff
@@ -709,6 +709,425 @@ def export_matrix_data(
...
+        model_name=None,
+        freq=0,
+        file_type="HSPICE",
+    ):
+        """Export matrix data.
+
+        Parameters
+        ----------
+        file_name : str
+            Full path to save the equivalent circuit to.
```
Comment (PipKat) [inline pyaedt/q3d.py:751]:
```suggestion
            Full path for saving the matrix data to.
```

File: pyaedt/q3d.py [hunk 5]
```diff
@@ -709,6 +709,425 @@ def export_matrix_data(
...
+        setup_name : str, optional
+            Setup name. The default value is ``None``, in which case the first
+            analysis setup is used.
+        sweep : str, optional
+            Solution frequency. The default is ``None``, in which case
+            the default adaptive is used.
+        variations : list or str, optional
+            Design variation. The default is ``None``, in which case the
+            current nominal variation is used.
+            If provided by user give the value as "{Name}:{Value}".
```
Comment (PipKat) [inline pyaedt/q3d.py:762]:
```suggestion
            current nominal variation is used. If you are provide a
            design variation, use the format ``{Name}:{Value}``.
```

File: pyaedt/q3d.py [hunk 6]
```diff
@@ -709,6 +709,425 @@ def export_matrix_data(
...
+        matrix_name : str, optional
+            Name of the matrix to display.
+            Default value is ``"Original"``.
```
Comment (PipKat) [inline pyaedt/q3d.py:765]:
```suggestion
            Name of the matrix to show. The default is ``"Original"```.
```

File: pyaedt/q3d.py (hunk 2 above)
Comment (gmalinve) [reply pyaedt/q3d.py:1008]: @MaxJPRey please see PR's description.
I would rather delete this part and always set include_cpp to False.

File: pyaedt/q3d.py (hunk 3 above)
Comment (gmalinve) [reply pyaedt/q3d.py:752]: good to know, thanks!

File: pyaedt/q3d.py (hunk 2 above)
Comment (MaxJPRey) [reply pyaedt/q3d.py:1008]: Indeed, I missed the description 😄 .
If you delete this section, maybe you can just had a comment (like the one in the PR description) to explain why it is always False.

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM

**Comment by PipKat** [comment]
@MaxJPRey Would you know why so many of my edits were ignored? Most of them are just to bring the text up to what our standards are for docstrings.

**Comment by gmalinve** [comment]
@PipKat I think on my side I approved all your suggestions :)

**Comment by PipKat** [comment]
@gmalinve Maybe? I see them as pending here on this page and on the Files page. Usually, there are no remaining comments.

**Comment by MaxJPRey** [comment]
@PipKat @gmalinve Everything seems ok. I can see the modifications in the main branch.

**Pull request merged**